quantumblacklabs/kedro-viz
# [KED-2385] Refactor row toggle/label CSS, and fix row hover icon bug

## Description

1. Instead of having just the _row.scss file and then filter overrides in _filter.scss, instead create one file for all the text label scss (both toggle/filter kinds), and one for the toggle-icon scss (both kinds)
2. Fix toggle-icon--kind-filter parent row-hover bug: The toggle-all icon would decrease in opacity on row hover.
3. Rename node-list row toggle `--kind-toggle` to `--kind-element` to make the distinction between the two elements a bit more clear. We can rename them again later if necessary but I think this makes things a lot easier to understand.
4. Refactor node-list group creation selector to improve readability

## Development notes

I discussed the naming of the two node-list row item kinds with @bru5 at length while working on this PR. We want to strike a balance between intuitiveness and extensibility, which is why we've avoided being too specific in the names of the two kinds (e.g. calling filters 'category', which might not work with future usages, or calling them 'blue-dot' and 'eye-icon', which is too dependent on the presentation).

## QA notes

Please double-check the node-list hover/focus/selection states and see that nothing has changed (except for the blue dot checked parent row-hover state, which has been fixed).

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
